do_with_in_base/src/lib.rs

```diff
@@ -1715,40 +1715,64 @@ fn arithmeticInternal<T: StartMarker + Clone, N: Copy + std::str::FromStr + std:
   };
 }
 
-/// A handler for basic arithmetic.
+/// Perform basic arithmetic calculations.
 /// 
 /// Because of a lack of type inference, you have to specify your desired return type at the start of any use of this handler.
-/// 
-/// Syntax is: `$output_specifier $command`
-///
-/// where `$output_specifier` is either a `$type`, or if you want to generate a non-suffixed literal result token, a `$type` suffixed with 'u'
-///
-/// where `$type` is one of `u8`, `i8`, `u16`, `i16`, `u32`, `i32`, `u64`, `i64`, `f32`, `f64`, `usize`, `isize`
+/// By default this will be a suffix-annotated literal (e.g. `5i8`, `1_i32`); append a `u` to the type name to return an unsuffixed literal.
+/// See [Rust by Example](https://doc.rust-lang.org/rust-by-example/primitives.html) for further reference.
 ///
-/// where `$command` is one of
-/// - `$val + $val`
-/// - `$val - $val`
-/// - `$val * $val`
-/// - `$val / $val`
-/// - `$val % $val // Remainder`
-/// - `$val | $val // Binary Or`
-/// - `$val ^ $val // Binary Xor`
-/// - `$val & $val // Binary And`
-/// - `not $val`
-/// - `size_of $type`
-/// 
-/// where `$val` is one of
-/// -  `$bracketed_command`
-/// -  `$number`
-///
-/// where `$bracketed_command` is one of
-/// - `( $command )`
+/// ## Supported Operators
+/// 
+/// | Operator | Operation | Notes |
+/// | -------- | --------- | ----- |
+/// | `+`       | Addition              |  |
+/// | `-`       | Subtraction           |  |
+/// | `*`       | Multiplication        |  |
+/// | `/`       | Division              |  |
+/// | `%`       | Modulo/Remainder      |  |
+/// | `\|`      | Bitwise OR            |  |
+/// | `^`       | Bitwise XOR           |  |
+/// | `&`       | Bitwise AND           |  |
+/// | `>`       | Bitwise right shift   | Note single `>` rather than more conventional `>>` |
+/// | `<`       | Bitwise left shift    | Note single `<` rather than more conventional `<<` |
+/// | `not`     | Bitwise NOT           | Unary. |
+/// | `size_of` | Size of type in bytes | Unary. Uses [https://doc.rust-lang.org/std/mem/fn.size_of.html] |
+/// 
```

Comment:
Perhaps a note that bitwise operators are not supported in float contexts, and negative literal numbers are not supported in unsigned context.

Reply:
(And generally literals that are out of bounds for their context won't work, eg any number greater than N::MAX or less than N::MIN for that N)